### PR TITLE
bunch of fixes

### PR DESCRIPTION
**Pull request by galaxyhaxz**
Everything should be bin exact up to multi.cpp. But I only checked function sizes and not full diff. `msgcmd.cpp` two functions have diff, but conveniently the size lines up so files after it have the right address.

I skipped `mainmenu.cpp` for now since DiabloUI will have to be rebuilt to link against it, since a new function was added to the UI.

Everything was tested against 1.08.

`logging.cpp` was copied from devilution, it's bin exact and will be finagled with later if needed.

### CODE REVIEW

File: Src/Diablo/Source/DIABLO.H [hunk 1]
```diff
@@ -97,7 +97,7 @@ void trace_fcn(const char * pszFcn);
 
 // EVEN version ID = retail version
 // ODD version ID = beta version
-#define VERSIONID	34
```
Comment (mewmew) [inline Src/Diablo/Source/DIABLO.H:100]: haha, so Hellfire was a beta version? :P

File: Src/Diablo/Source/MISSILES.CPP [hunk 2]
```diff
@@ -403,6 +403,12 @@ int GetDirection16(int x1, int y1, int x2, int y2)
 **-----------------------------------------------------------------------*/
 void DeleteMissile(int mi, int i)
 {
+	if (missile[mi]._mitype == MIT_MANASHIELD) {
+		int id = missile[mi]._misource;
+		if (id == myplr) NetSendCmd(TRUE, CMD_REMSHIELD);
```
Comment (mewmew) [inline Src/Diablo/Source/MISSILES.CPP:408]: yay, got use for the new enum defined in #3 :)

File: Src/Diablo/Source/MSG.CPP [hunk 3]
```diff
@@ -3847,6 +3847,49 @@ static DWORD On_CHEAT_SPELL_LEVEL( const TCmd* pCmd, int pnum ) {
...
+**-----------------------------------------------------------------------*/
+static DWORD On_NOVA( const TCmd* pCmd, int pnum )
+{
+	if (gbBufferMsgs != BUFFER_ON && currlevel == plr[pnum].plrlevel && pnum != myplr) {
+		DROPLOG("CMD_NOVA(%d)\n",pnum);
+		TCmdLoc *p = (TCmdLoc *)pCmd;
+		ClrPlrPath(pnum);
+		plr[pnum]._pSpell = SPL_NOVA;
+		plr[pnum]._pSplType = 4;
+		plr[pnum]._pSplFrom = 3;
```
Comment (mewmew) [inline Src/Diablo/Source/MSG.CPP:3860]:
```suggestion
		plr[pnum]._pSplFrom = SPL_FROMSB;
```

File: Src/Diablo/Source/MSG.CPP [hunk 4]
```diff
@@ -3847,6 +3847,49 @@ static DWORD On_CHEAT_SPELL_LEVEL( const TCmd* pCmd, int pnum ) {
...
+/*-----------------------------------------------------------------------**
+**-----------------------------------------------------------------------*/
+static DWORD On_NOVA( const TCmd* pCmd, int pnum )
+{
+	if (gbBufferMsgs != BUFFER_ON && currlevel == plr[pnum].plrlevel && pnum != myplr) {
+		DROPLOG("CMD_NOVA(%d)\n",pnum);
+		TCmdLoc *p = (TCmdLoc *)pCmd;
+		ClrPlrPath(pnum);
+		plr[pnum]._pSpell = SPL_NOVA;
+		plr[pnum]._pSplType = 4;
```
Comment (mewmew) [inline Src/Diablo/Source/MSG.CPP:3859]:
```suggestion
		plr[pnum]._pSplType = SPT_NONE;
```

File: Src/Diablo/Source/MSG.CPP [hunk 5]
```diff
@@ -3847,6 +3847,49 @@ static DWORD On_CHEAT_SPELL_LEVEL( const TCmd* pCmd, int pnum ) {
...
+		plr[pnum]._pSplFrom = 3;
+		plr[pnum].destAction = 12;
```
Comment (mewmew) [inline Src/Diablo/Source/MSG.CPP:3861]:
```suggestion
		plr[pnum].destAction = PCMD_SPELL;
```

File: Src/Diablo/Source/DIABLO.H (hunk 1 above)
Comment (galaxyhaxz) [reply Src/Diablo/Source/DIABLO.H:100]: Hellfire used 34, which is an even number so retail. 1.08 used 40, 1.09 used 42.

Comment (mewmew) [reply Src/Diablo/Source/DIABLO.H:100]: hehe, for some reason, my mind only read the first digit, `3`, well that's not even :)

**Pull request merged**